app/src/sharedTest/java/org/oppia/app/testing/InputInteractionViewTestActivityTest.kt

```diff
@@ -222,4 +223,199 @@ class InputInteractionViewTestActivityTest {
     onView(withId(R.id.test_fraction_input_interaction_view)).check(matches(isDisplayed()))
       .check(matches(withText("9/5")))
   }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withNoInputText_hasCorrectPendingAnswerType() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_number_with_units_input_interaction_view) as NumberWithUnitsInputInteractionView
+      val interactionObject = textAnswerRetriever.getPendingAnswer()
+      assertThat(interactionObject).isInstanceOf(InteractionObject::class.java)
+      assertThat(textAnswerRetriever.getPendingAnswer().real).isWithin(1e-5).of(0.0)
+      assertThat(interactionObject.numberWithUnits.fraction.denominator).isEqualTo(0)
+      assertThat(interactionObject.numberWithUnits.fraction.numerator).isEqualTo(0)
+      assertThat(interactionObject.numberWithUnits.fraction.wholeNumber).isEqualTo(0)
+      assertThat(interactionObject.numberWithUnits.fraction.isNegative).isEqualTo(false)
+      assertThat(interactionObject.numberWithUnits.unitOrBuilderList).isEmpty()
+    }
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withInputtedNegativeWholeNumberText_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
```

Comment:
You have already created this `activityScenario` globally, use that directly here and everywhere else.

Reply:
Need to work on configuration change for that i need to use rules and and launch using that. That need to be done once other dependences are cleared .